code/controllers/subsystems/ghostroles.dm

```diff
@@ -0,0 +1,126 @@
+#define UIDEBUG
+
+/var/datum/controller/subsystem/ghostroles/SSghostroles
+
+/datum/controller/subsystem/ghostroles
+	name = "Ghost Roles"
+	flags = SS_NO_FIRE
+	init_order = SS_INIT_MISC_FIRST
+
+	var/list/spawnpoints = list() //List of the available spawnpoints by spawnpoint type
+		// -> type 1 -> spawnpoint 1
+		//           -> spawnpoint 2
+
+	var/list/spawners = list() //List of the available spawner datums
+
+/datum/controller/subsystem/ghostroles/Recover()
+	src.spawnpoints = SSghostroles.spawnpoints
+	src.spawners = SSghostroles.spawners
+
+/datum/controller/subsystem/ghostroles/New()
+	NEW_SS_GLOBAL(SSghostroles)
+	for(var/spawner in subtypesof(/datum/ghostspawner))
+		var/datum/ghostspawner/G = new spawner
+		//Check if we hae name, short_name and desc set
+		if(!G.short_name || !G.name || !G.desc)
+			continue
```

Comment:
I'd personally add a `crash_with("Bad ghost spawner.")` here. The idea is that these will get created during unit testing and emitting a runtime will cause the unit tests to fail.

Thus, we detect a bad ghost spawner right off the bat! At least, if all ghost spawners are meant to adhere to these requirements.

Reply:
As discussed, that is not possible due to child objects used to organize the spawners.